go.mod

```diff
@@ -9,6 +9,7 @@ require (
 	github.com/onsi/ginkgo v1.14.0
 	github.com/onsi/gomega v1.10.1
 	github.com/pborman/uuid v1.2.0
+	github.com/prometheus/client_golang v1.11.1 // indirect
```

Comment:
I am curious where is `github.com/prometheus/client_golang` lib used in this project? I could not find any code referencing prometheus/client_golang

Reply:
o, it's referenced by https://github.com/kubernetes-csi/csi-driver-nfs/blob/0a08bf7c377c5459e48bd2ca014fca4eda7b58c1/vendor/k8s.io/component-base/metrics/legacyregistry/registry.go#L22-L23, so we should upgrade `component-base` in go.mod

Reply:
set `indirect` dependency is not ideal way in long term maintenance

Reply:
the vendor change only applies in k8s 1.24: https://github.com/kubernetes/kubernetes/pull/108328

Reply:
You are correct, it's indirect. The idea is to remove the explicit `require` once dependencies bump it themselves.
This patch has the advantage that it can be easily backported. Shall we merge, or wait for a k8s bump? I have no strong opinion.